test/resources/output/NullHeader.yml

```diff
@@ -4,6 +4,7 @@ info:
   version: 1.0.0
 servers:
   - url: http://{{url}}
+    variables: {}
```

Comment:
I went back and forth on this one on whether or not it should include `url` in here as a server variable. Here at [ReadMe](https://readme.com) when we have these situations in our OpenAPI engine where there's a missing server variable in `variables` we set the `default` to be the name of the variable. So like in this case this would look like:

```
variables: { url: { default: 'url' } }
```

Not sure if that's how you want this to be though, but an empty `variables` object like this still passes OpenAPI spec validation.